### ISSUE TITLE
iOS Resume Playback Problem

### ISSUE THREAD

**Issue by slyfoxgh** [advplyr/audiobookshelf-app#844]
1. Turn on ABS.
2. Lock the screen.
3. Wait about 6 minutes without playing.
4. Lock screen player disappears.

The same issue occurs while using the app in the car. After about five minutes of pausing, there is no way to resume playback unless you open the app and navigate to the file you want to play.

**Comment by Drudoo** [comment]
You don't even have to wait 6 min, usually 1-2 min is enough for the player to disappear.

**Comment by xstefanx** [comment]
I also have this issue, usually about one minute or so and the app seems to close entirely. I’d seen mention of this bug in other issues that are marked as closed now. #770 has some links to previous issues as well. 

I use the app with carplay a lot and while the app doesn’t have full functionality of carplay, being able to pause and resume from the now playing app is at least safer than having to pull out the phone and open the app and specific book again and then hit play from there.

**Comment by xstefanx** [comment]
I actually noticed today that this happens even if the app is open behind the locked screen. I had left the app running and the screen timed out. I paused from the car stereo and noticed it had timed out a few seconds later when I went to resume it. Upon unlocking my phone I could see the app for a brief second and then it closed. Seems a lot like the app is crashing while not active on screen and paused.

**Comment by nichwall** [comment]
What iOS version are y'all running? Are you running the iOS 17 beta?

**Comment by advplyr** [comment]
I haven't been able to reproduce this with iOS 16.6.1 yet, still testing different scenarios

**Comment by xstefanx** [comment]
I’m on 16.0 I can try updating to see if that changes anything.

**Comment by xstefanx** [comment]
Can confirm it still happens to me on 16.6.1

**Comment by DavidCo23** [comment]
This was happening to me a lot as well, even after the most recent update to the app. Looking at the server logs it was getting spammed with "[MeController] syncLocalMediaProgress invalid local media progress object no library item [object Object]" repeatedly. I tried getting rid of a lot of my locally downloaded audiobooks and it didn't seem to help. I ended up uninstalling/reinstalling the app and that seems to have fixed the issue for now. Able to resume audio after several minutes of being paused on a locked screen again.

**Comment by xstefanx** [comment]
Uninstall and reinstall did indeed fix the app crashing while paused. 

Very strange and I’m sure it’ll make it near impossible to troubleshoot if it’s even an issue with the app itself. Could be an iOS bug as well.

Thanks for the tip!

**Comment by advplyr** [comment]
Thanks for testing this. 
It seems to be the case that on iOS if you had downloaded media on server v2.2.23 or earlier and listened offline then you may need to clear data in the app.

**Comment by xstefanx** [comment]
> Thanks for testing this. It seems to be the case that on iOS if you had downloaded media on server v2.2.23 or earlier and listened offline then you may need to clear data in the app.

I've actually never downloaded content at all. I've only streamed.

**Comment by advplyr** [comment]
@xstefanx Ah yeah, I forgot that the iOS app will store progress syncs locally even for media not downloaded if the progress syncs fails. Then it will try to sync it the next time any media gets played.

This also explains why it's a small percentage of iOS users coming across this.

**Comment by tinect** [comment]
same here

**Comment by cybergurke** [comment]
I can also confirm the issue

**Comment by watsonbox** [comment]
I too was having this issue - player disappearing a app screen refreshing after a few minutes of pause - and can confirm that reinstalling the app fixed it.

**Comment by lerndmina** [comment]
I am having this issue on iOS 17.0.2 (iPhone 15 Pro)

After pausing playback and waiting about 10-60 seconds the app disappears from the playback controls in the control centre and the dynamic island/lock screen. Upon re opening the app it sends me to the main page and once I resume playback it resumes at the correct point.

**Comment by advplyr** [comment]
@lerndmina Have you tried reinstalling the app?

**Comment by lerndmina** [comment]
> @lerndmina Have you tried reinstalling the app?

Yeah I just reinstalled today. And it seems to still be happening.

**Comment by mym0T** [comment]
I'm having this issue as well. Server version: 2.4.4
iPhone 12 Pro on iOS 17.0.3
App Version: 0.9.66-beta
I tried re-installing the app, but it didn't resolve the issue for me either.

The only thing I see in the logs is connect and disconnect messages.

```
{"timestamp":"2023-10-06 00:00:24","source":"Logger.js:114","message":"[SocketAuthority] Socket H-oLFVtn_apGr2vyAAAr disconnected from client "username" after 109653ms (Reason: transport close)","levelName":"INFO","level":2}
```

**Comment by themaz** [comment]
On iOS beta 17.1, same issue. Uninstalled, restarted, and reinstalled the app but it did not help. 

With Prologue (audiobook app that uses plex as a server, I don’t experience this behavior)

**Comment by oswalt0** [comment]
Do you log into the server via local only? I was having the same problem, it seems to be an "offline" issue, until I installed Tailscale and now it only does it on my Android that is not going through Tailscale to the library. If you watch the sync at the top when you first start your book as soon as its done refreshing it seems to press pause and then you have to do press play again. Also if you are not connected to your library via an internet connection it also seems to force close the app when you press pause via a Bluetooth connection. I have been able to consistently reproduce the problem if I go back to my local IP and use offline.

**Comment by lerndmina** [comment]
> Do you log into the server via local only?
Idk about others but I have a VPN to my server that's always on my phone so I can say that this issue happens regardless of whether I am connected to the server or not.

**Comment by tinect** [comment]
my server is online.

**Comment by lerndmina** [comment]
Hey. Just a quick note. I don't know what has happened but this is happening less frequently on my iOS device. So it's even harder to pin down a specific workflow or cause to this issue 


But it is getting better at least for me. 

I don't know if it has anything to do with the fact that I moved from hosting using the apt package to
Docker. 🤷‍♂️ probably not.

**Comment by ZeroNoir51** [comment]
It fixed itself for me as well

**Comment by advplyr** [comment]
Users in discord confirmed that this issue is fixed in the iOS release today v0.9.67-beta
